Library/Homebrew/rubocops/lines.rb

```diff
@@ -210,15 +221,36 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
 
             find_instance_method_call(body_node, "Utils", unsafe_command) do |method|
               unless test_methods.include?(method.source_range)
+                @problem_cause = "unsafe"
                 problem "Use `Utils.safe_#{unsafe_command}` instead of `Utils.#{unsafe_command}`"
               end
             end
           end
+
+          # Separate shell variables to a hash (e.g. `Utils.safe_popen_read({ "SHELL" = "bash" }, "foo")`)
+          popen_commands.each do |command|
+            find_instance_method_call(body_node, "Utils", command) do |method|
+              next unless match = regex_match_group(parameters(method).first, /^([^"' ]+)=([^"' ]+)(?: (.*))?$/)
+
+              @problem_cause = "variables"
+              @fixed_version = "{ \"#{match[1]}\" => \"#{match[2]}\" }, \"#{match[3]}\""
+              good_args = "Utils.#{command}(#{@fixed_version})"
+
+              problem "Use `#{good_args}` instead of `#{method.source}`"
+            end
+          end
         end
 
         def autocorrect(node)
           lambda do |corrector|
-            corrector.replace(node.loc.selector, "safe_#{node.method_name}")
+            case @problem_cause
```

Comment:
This makes me wonder if it's be a better fit in two separate cops/autocorrects. Thoughts?

Reply:
I wasn't totally sure. This does feel like a hacky way to do it but I wasn't sure what the best thing to do is. My thinking for keeping them together is that they both deal with shell commands (specifically the popen_commands) so it would reduce unnecessary clutter in the file

By the way, I'm assuming that "cop" refers to each individual `class Foo < FormulaCop`, correct?

Reply:
Actually, I just looked at the file more closely and I think it does make sense for them to be separate. For some reason, I was thinking that each of the other cops in the file was long and covered many things but that doesn't actually seem to be true. I'll go ahead and separate it out.

Reply:
> By the way, I'm assuming that "cop" refers to each individual `class Foo < FormulaCop`, correct?

Yup!



> I'll go ahead and separate it out.

❤️